donut/modules/arcfeedback/routes.py

```diff
@@ -0,0 +1,132 @@
+    return flask.redirect(flask.url_for('arcfeedback.arcfeedback'))
+
+
+# api endpoint with all visible data
+@blueprint.route('/1/arcfeedback/view/<uuid:id>')
+def arcfeedback_api_view_complaint(id):
+    if not (helpers.get_id(id)):
+        return flask.render_template("404.html")
+    complaint_id = helpers.get_id(id)
+    #pack all the data we need into a dict
+    return flask.jsonify(helpers.get_all_fields(complaint_id))
+
+
+# view a complaint
+@blueprint.route('/arcfeedback/view/<uuid:id>')
+def arcfeedback_view_complaint(id):
+    if not (helpers.get_id(id)):
+        return flask.render_template("404.html")
+    complaint_id = helpers.get_id(id)
+    complaint = helpers.get_all_fields(complaint_id)
+    complaint['uuid'] = id
+    return flask.render_template('complaint.html', complaint=complaint)
+
+
+# add a message to this post
+@blueprint.route('/1/arcfeedback/add/<uuid:id>', methods=['POST'])
+def arcfeedback_add_msg(id):
+    complaint_id = helpers.get_id(id)
+    if not complaint_id:
+        flask.abort(400)
+        return
+    fields = ['message', 'poster']
+    data = {}
+    for field in fields:
+        data[field] = flask.request.form.get(field)
+    if data['message'] == "":
+        flask.abort(400)
+        return
+    helpers.add_msg(complaint_id, data['message'], data['poster'])
+    return flask.jsonify({
+        'poster': data['poster'],
+        'message': data['message']
+    })
+
+
+# allow arc members to see a summary
+@blueprint.route('/arcfeedback/view/summary')
+def arcfeedback_view_summary():
+    #authenticate
+
+    #get a list containing data for each post
+    complaints = helpers.get_new_posts()
+    #add links to each complaint
+    for complaint in complaints:
+        complaint['link'] = helpers.get_link(complaint['complaint_id'])
+    return flask.render_template('summary.html', complaints=complaints)
+
+
+# mark a complaint read
+@blueprint.route('/1/arcfeedback/markRead/<uuid:id>')
+def arcfeedback_mark_read(id):
+    #authenticate
+
+    complaint_id = helpers.get_id(id)
+    if helpers.mark_read(complaint_id) == False:
+        flask.abort(400)
+        return
+    else:
+        return 'Success'
+
+
+# mark a complaint unread
+@blueprint.route('/1/arcfeedback/markUnread/<uuid:id>')
+def arcfeedback_mark_unread(id):
+    #authenticate
+
+    complaint_id = helpers.get_id(id)
+    if helpers.mark_unread(complaint_id) == False:
+        flask.abort(400)
+        return
+    else:
+        return 'Success'
+
+
+# add an email to this complaint
+@blueprint.route('/1/arcfeedback/<uuid:id>/addEmail', methods=['POST'])
+def arcfeedback_add_email(id):
+    complaint_id = helpers.get_id(id)
+    emails = flask.request.form.get('email')
+    emails = emails.split(',')
+    emails = [email.strip() for email in emails]
+    success = helpers.add_email(complaint_id, emails)
+    if not success: return "Failed to add email(s)"
+    return "Success!"
```

Comment:
return the same complaint page to bring the user back to where they added the email